x/oracle/keeper/historic_price.go

```diff
@@ -106,50 +80,126 @@ func (k Keeper) GetMedianDeviation(
 	return decProto.Dec, nil
 }
 
-// WithinMedianDeviation returns whether or not a given price of a given
-// denom is within the Standard Deviation around the Median.
-func (k Keeper) WithinMedianDeviation(
+// WithinHistoricMedianDeviation returns whether or not the current price of a
+// given denom is within the latest stamped Standard Deviation around
+// the Median.
+func (k Keeper) WithinHistoricMedianDeviation(
 	ctx sdk.Context,
 	denom string,
-	price sdk.Dec,
 ) (bool, error) {
-	median, err := k.GetMedian(ctx, denom)
-	if err != nil {
-		return false, err
+	// get latest median
+	medians := k.HistoricMedians(ctx, denom, 1)
+	if len(medians) == 0 {
+		return false, sdkerrors.Wrap(types.ErrNoMedian, fmt.Sprintf("denom: %s", denom))
+	}
+	median := medians[0]
+
+	// get latest historic price
+	prices := k.historicPrices(ctx, denom)
+	if len(prices) == 0 {
+		return false, sdkerrors.Wrap(types.ErrNoHistoricPrice, fmt.Sprintf("denom: %s", denom))
 	}
+	price := prices[0]
```

Comment:
if we want only the latest price, why do we deserialize all of them? Please update, (you can use reverse iterator) to only fetch max one price. (it's possible that there will be zero)

Reply:
Update to iterate until a start block like the medians iterator 👍

Reply:
I don't see the update

Reply:
Im calling for only the latest historic stamp now rather than calling for all of them; `prices := k.historicPrices(ctx, denom, 1)` the "1" parameter being numstamps